src/Interpreters/formatWithPossiblyHidingSecrets.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+#include "Access/ContextAccess.h"
+#include "Interpreters/Context.h"
+
+namespace DB
+{
+struct SecretHidingFormatSettings
+{
+    // We can't store const Context& as there's a dangerous usage {.ctx = *getContext()}
+    // which is UB in case getContext()'s return ptr is the only one holding the object
+    const ContextPtr & ctx;
+    const IAST & query;
+    size_t max_length = 0;
+    bool one_line = true;
+};
+
+inline String format(const SecretHidingFormatSettings & settings)
```

Comment:
to general name, let's name it as file name?

Reply:
I believe this function's name should serve as a convenience wrapper, so as a) it's used rarely b) c++ code uses overloads, if you don't mind, let's leave this as is